assets/scripts/fe/src/app/commons/services/history/standup.service.ts

```diff
@@ -1,20 +1,118 @@
+import * as _ from 'lodash';
+
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 
 import { urlsafe, queryparams } from '../../utils/http.utils';
-import { HISTORY_STANDUP } from '../../constants/api.constants';
-import { Standup } from '../../models/history.models';
-
+import { HISTORY_STANDUP, ACCOUNTING_PROJECT_DETAILS, HISTORY_STANDUP_WEEKLY } from '../../constants/api.constants';
+import { ProjectDetail } from '../../../commons/models/project-detail.models';
 
 @Injectable({
   providedIn: 'root'
 })
 export class StandupService {
 
+  // contains the list of weekly reports
+  public q = [];
+  // boolean to check if currently getting new weekly report
+  public fetching = false;
+  // boolean to check if all data are loaded
+  public all_loaded = false;
+  // url parameter for the next page
+  public qparams = {
+    page: 1
+  };
+  // set so that when side panel is called it will not reload all data
+  public noreload = false;
+
+  //variables for getting and setting dates
+  public dt :Date;
+  public currentWeekDay :number;
+  public lessDays :number;
+  public wkStart :Date;
+  public wkEnd :Date;
+
+  public date_test = {
+    'year':0,
+    'month':0
+  };
+
+  // contains the project details
+  public projectDetails = new ProjectDetail;
+
   constructor(
     private http: HttpClient
   ) { }
 
+  getWeeklyReport(id) {
+    // toggle fetching to true to prevent multiple
+    // similar request to overload the server.
+    this.fetching = true;
+    // set url with project id
+    let url = urlsafe(HISTORY_STANDUP_WEEKLY, id)
+    // set url with date
+
+    let weekStart = this.wkStart.getFullYear() + "-" + (this.wkStart.getMonth() + 1) + "-" + this.wkStart.getDate()
+
+    this.http.get(urlsafe(url, weekStart) + queryparams(this.qparams))
```

Comment:
bad url format.

it should be something like this:
`/api/history/standups/?date_start=2019-05-06&project_id=1`